src/ctl.c

```diff
@@ -2983,6 +2985,61 @@ arena_i_retain_grow_limit_ctl(tsd_t *tsd, const size_t *mib,
 	return ret;
 }
 
+/*
+ * When writing, newp should point to a char array storing the name to be set.
+ * A name longer than ARENA_NAME_LEN will be arbitrarily cut. When reading,
+ * oldp should point to a char array whose length is no shorter than
+ * ARENA_NAME_LEN or the length of the name when it was set.
+ */
+static int
+arena_i_name_ctl(tsd_t *tsd, const size_t *mib, size_t miblen,
+    void *oldp, size_t *oldlenp, void *newp, size_t newlen) {
+	int ret;
+	unsigned arena_ind;
+	char *name = NULL;
```

Comment:
nit: no need to initialize here. We try avoiding double initialization, meaning that it should be explicitly initialized down the path its needed. This way the compiler can warn us if there's any possibility of access w/o init (which won't be possible if it's simply initialized at the beginning. The `ret` in this function is a good example of following this principle.

Reply:
removed the initialization